src/main/thrift/parquet.thrift

```diff
@@ -316,7 +317,7 @@ struct BsonType {
  * LogicalType annotations to replace ConvertedType.
  *
  * To maintain compatibility, implementations using LogicalType for a
- * SchemaElement must also set the corresponding ConvertedType from the
+ * SchemaElement should also set the corresponding ConvertedType from the
```

Comment:
sorry for the belated comment by why the change from must to should?

Reply:
I also believed that setting ConvertedType was MUST for forward compatibility reasons. Can we please discuss on the mailing list?

Reply:
Logical types were introduced 3.5 years go, so it seemed reasonable to relax the requirement. Do you expect some Parquet readers in the wild to still not understand them?
It's easy, for every converted type you support, to also support the corresponding logical type. It's only a small amount of logic in the decoding path, and it's not performance-critical.

Reply:
That said, feel free to spawn a discussion on the ML.

Reply:
If you want to make this change, I think it needs to be voted on. Knowing the motley state of Parquet implementations in enterprise deployments I would be uncomfortable writing files that don't have the field.

Reply:
Well, given the state, even today, of inter-implementation compatibility, I think anyone using a 2017-era implementation would run into lots of issues when reading files produced by other implementations, regardless of whether logical types are used.

Reply:
That said, if people feel strongly about it, I can simply revert this one-word change, because I don't think there's any point organizing a vote on it.

Reply:
It's a change with forward compatibility implications, so either we revert or we discuss/vote, but we can't leave it like this. Impala for example only started to get support for LogicalType in 2019 (and this is only one of the mainstream independent implementations), so we'd be making a bet about how quickly updates are rolling out to all of the corners of the world

https://github.com/apache/impala/commit/0906e0817ce2301d7e20f355b334861f0232f16f